File: app/src/main/java/teamenglify/englify/Model/ExerciseChapterPart.java
package teamenglify.englify.Model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.LinkedList;

/**
 * Created by keith on 29-Mar-17.
 */

public class ExerciseChapterPart implements Serializable{
    public String name;
    public String question;
    public ArrayList<String> answer;
    public ArrayList<ArrayList<String>> choices;  //Outer array is a list for each "blank" in the question. Inner array is a list of choices for "blank" in question at index n.
    public String imageURL;
    public String audioURL;

    public ExerciseChapterPart(String name) {
        this.name = name;
    }

    public ExerciseChapterPart(String name, String audioURL, String imgURL) {
        this.name = name;
        this.imageURL = imageURL;
        this.audioURL = audioURL;
    }
}

File: app/src/main/java/teamenglify/englify/Model/ExerciseChapterPart.kt
package teamenglify.englify.Model

import io.realm.RealmList
import io.realm.RealmObject

open class ExerciseChapterPart: RealmObject() {
    lateinit var name: String
    lateinit var question: String
    var answer = RealmList<String>()
    var choices = RealmList<Choice>()
    lateinit var translation: String
    lateinit var imageUrl: String
    lateinit var audioUrl: String
}